collector/lib/NetworkStatusNotifier.cpp

```diff
@@ -300,12 +303,40 @@ sensor::NetworkConnectionInfoMessage* NetworkStatusNotifier::CreateInfoMessage(c
 }
 
 void NetworkStatusNotifier::AddConnections(::google::protobuf::RepeatedPtrField<sensor::NetworkConnection>* updates, const ConnMap& delta) {
+  // Update the rate limiter with new bounds, in case the runtime
+  // configuration has been updated
+  connections_rate_limiter_.SetBurstSize(config_.PerContainerRateLimit());
+  connections_rate_limiter_.SetCapacity(config_.PerContainerRateLimit());
```

Comment:
Done, and I've also switched this around to create a new rate limiter directly in this function - due to the way we're rate limiting, we actually don't need the caching part of the RateLimitCache, so a new instance per call to AddConnections should be sufficient. There's certainly an argument that we could create a non-caching version of RateLimitCache, and I'm not completely opposed to that if it would be clearer. WDYT?

Reply:
I kinda did it anyway, since it's pretty simple: https://github.com/stackrox/collector/pull/1945/commits/1ddc8aad57fad13aee87d511f8259c878fdcdb74